homeassistant/components/intellifire/binary_sensor.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+from . import IntellifireDataUpdateCoordinator
+
+"""Support for Intellifire Binary Sensors."""
+
+from typing import Any
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DOMAIN,
+    HOT,
+    INTELLIFIRE_BINARY_SENSORS,
+    PILOT,
+    POWER,
+    THERMOSTAT,
+    TIMER,
+)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up a Intellifire On/Off Sensor"""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities = [
+        IntellifireBinarySensor(
+            coordinator=coordinator, entry_id=entry.entry_id, description=description
+        )
+        for description in INTELLIFIRE_BINARY_SENSORS
+    ]
+    async_add_entities(entities)
+
+
+class IntellifireBinarySensor(CoordinatorEntity, BinarySensorEntity):
+    """A semi generic wrapper around Binary Sensor entiteis for Intellifire"""
+
+    def __init__(
+        self,
+        coordinator: IntellifireDataUpdateCoordinator,
+        entry_id,
+        description: BinarySensorEntityDescription,
+    ):
+        super().__init__(coordinator)
+        self.entity_description = description
+        self.coordinator = coordinator
+        self._entry_id = entry_id
+        self._attrs: dict[str, Any] = {}
+
+        # Set the Dispaly name the User will see
+        self._attr_name = f"{coordinator.intellifire_name} Fireplace {description.name}"
+        self._attr_unique_id = (
+            f"Intellifire_{coordinator.safe_intellifire_name}_{description.key}"
```

Comment:
This doesn't look like it meets the [requirements for a unique ID](https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#unique-id-requirements); in particular the comment on `safe_intellifire_name` suggests that it is changeable by the user.

Reply:
Thanks for this catch! I had tried to use the serial number at one point ... working on making that change now.